radicle-cli/src/commands/patch/checkout.rs

```diff
@@ -0,0 +1,41 @@
+use crate::terminal as term;
+use radicle::cob::patch::{self, PatchId};
+use radicle::git;
+use radicle::profile::Profile;
+use radicle::storage::git::Repository;
+
+pub fn run(
+    storage: &Repository,
+    profile: &Profile,
+    git_workdir: &git::raw::Repository,
+    patch_id: &PatchId,
+) -> anyhow::Result<()> {
+    let patches = patch::Patches::open(profile.public_key, storage)?;
+    let Some(patch) = patches.get(patch_id)? else {
+        anyhow::bail!("Patch `{patch_id}` not found")
+    };
+
+    let spinner = term::spinner("Performing patch checkout...");
+
+    // Getting the patch obj!
+    let patch_head = patch.head();
+    let obj = git_workdir.revparse_single(&patch_head.to_string())?;
```

Comment:
Hmm maybe instead of this and then `as_commit`, we could use https://docs.rs/git2/latest/git2/struct.Repository.html#method.find_commit directly using the `oid` from `patch.head()`